### PR TITLE
AP_Logger: allow for retry of log open with LOG_DISARMED=1

### PR DESCRIPTION
**Pull request by tridge**
if booting without a microSD and with LOG_DISARMED=1 then this allows
for a retry of opening the log every 5s
ping @WickedShell

### CODE REVIEW

File: libraries/AP_Logger/AP_Logger_File.cpp [hunk 1]
```diff
@@ -135,6 +138,17 @@ void AP_Logger_File::periodic_1Hz()
...
     }
 
+    if (_write_fd == -1 && _read_fd == -1 &&
+        logging_enabled() &&
+        !recent_open_error() &&
+        !hal.util->get_soft_armed()) {
+        // retry logging open. This allows for booting with
+        // LOG_DISARMED=1 with a bad microSD or no microSD. Once a
+        // card is inserted then logging starts
+        _initialised = true;
```
Comment (peterbarker) [inline libraries/AP_Logger/AP_Logger_File.cpp]: Forcing this doesn't seem entirely safe at first glance - have we necessarily got a valid _writebuf pointer?

Comment (tridge) [reply libraries/AP_Logger/AP_Logger_File.cpp]: I think it is protected by the checks for _write_fd and _read_fd being -1

**Comment by peterbarker** [comment]
I've rebased this to resolve conflicts.

I've also changed this to NOT reset _initialised as part of retrying the log open; recent changes in the Logger init functions to move directory creation mean the only failures in there are permanent and debilitating things we can't ignore by forcing the initialised flag.

**Comment by peterbarker** [comment]
I've tested this on a PixRacer, being variously nasty to it.

Looks good - apart from triggering a pre-existing problem with the watchdog on log erase; we may need to push some expected delays up.

We no longer set `_initialised` to false on what might be transient errors.

**Comment by peterbarker** [comment]
I've rebased this again, fixing a conflict in the header file.

I intend to merge this after CI passes.

**Comment by WickedShell** [comment]
Thanks @peterbarker  I've been cherry picking this onto all my plane builds for more then a year now, so nice to see this get in. :)

**Pull request merged**